cognitedata/reveal
# (feat) Adding unit to measurments + fixing picking on the measurment views + Renaming

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:
1. Rename Tooltip to TranslateKey 
2.  Add unique index in  BaseCommand  to forces rerender in react for special rare cases.
3. Added CopyToClipboardCommand and  DeleteDomainObjectCommand, and used them in the DomainObjectPanel
4. Added ToogleMetricUnitsCommand for switching between metric and none metric units.
5. Added more comments around in the base of the architecture
6. Added unitSystem to the RootDomainObject , so units are handled one place
7. Rename NumberType  -> Quantity
8.  Refactor of PanelInfo to hadle the new unit system
9. Added class UnitSystem  to handle very basic unit system. This should be extended in the future.
10. Moved class ClosestGeometryFinder from Fusion. This is not deleted on the fusion side, but should when Annotations are using the new architecture
11. Added unit change in Changes and used when it was necessary
12. Use unit system in the measurement views and in the DomainObjectPanel'
13. Implement intersectIfCloser in MeasureLineView because the tree.js implementation did not work properly. 
14. Update the example code with the new unit system
15. Some refactoring in the CommandButtons.cs
16. Remove canBeChecked in commands. Not used.


TODO: The axis no not yet understand the unit system. It should. Will take a couple of hours.

## How has this been tested? :mag:

See below

## Test instructions :information_source:

1. Use the measurement tool to create some measurements
2. Note the new command with the measurement icon
3. Toggle this on and off. See the difference

![image](https://github.com/cognitedata/reveal/assets/35219649/1cd77c26-013b-4023-b30f-20108f8a31d6)

Icons and the placement is discussed with Designer.

## Checklist :ballot_box_with_check:
